packages/salesforcedx-vscode-soql/src/commonUtils.ts

```diff
@@ -6,23 +6,48 @@
  */
 
 import * as path from 'path';
-import { TextDocument, workspace, WorkspaceFolder } from 'vscode';
+import * as vscode from 'vscode';
+import { channelService } from './channel';
+import { telemetryService } from './telemetry';
 
-export function getDocumentName(document: TextDocument): string {
+export function getDocumentName(document: vscode.TextDocument): string {
   const documentPath = document.uri.fsPath;
   return path.basename(documentPath) || '';
 }
 
-function hasRootWorkspace(ws: typeof workspace = workspace) {
+function hasRootWorkspace(ws: typeof vscode.workspace = vscode.workspace) {
   return ws && ws.workspaceFolders && ws.workspaceFolders.length > 0;
 }
 
-function getRootWorkspace(): WorkspaceFolder {
+function getRootWorkspace(): vscode.WorkspaceFolder {
   return hasRootWorkspace()
-    ? workspace.workspaceFolders![0]
-    : ({} as WorkspaceFolder);
+    ? (vscode.workspace.workspaceFolders as vscode.WorkspaceFolder[])[0]
+    : ({} as vscode.WorkspaceFolder);
 }
 
 export function getRootWorkspacePath(): string {
   return getRootWorkspace().uri ? getRootWorkspace().uri.fsPath : '';
 }
+
+export function showError(kind: string, err: string): void {
+  vscode.window.showErrorMessage(`${kind}:  ${err}`);
+}
+
+export async function trackError(kind: string, err: string): Promise<void> {
+  try {
+    await telemetryService.sendException(
+      `soql_error_${kind.toLocaleLowerCase()}`,
+      err
+    );
+  } catch (err) {
+    channelService.appendLine(`soql_error_telemetry:  ${err.toString()}`);
+  }
+}
+
+export async function showAndTrackError(
+  kind: string,
+  err: string
+): Promise<void> {
+  showError(kind, err);
```

Comment:
We should also make sure the errors we show customers are useful. Here are a couple of examples of error messages that need to be changed cc @ekapner 

<img width="477" alt="Screen Shot 2020-12-08 at 11 26 45 AM" src="https://user-images.githubusercontent.com/1041842/101531815-9549c380-3948-11eb-8069-b774cbdd1cba.png">

<img width="465" alt="Screen Shot 2020-12-08 at 11 12 49 AM" src="https://user-images.githubusercontent.com/1041842/101531840-9d096800-3948-11eb-8c77-7d5f53724f57.png">

Reply:
@dehru - feel free to ping me and we can meet to walk through error message text.

Reply:
@lcampos - How did you trigger these errors?  I've had a hard time triggering most of these.

Reply:
@dehru if I remember correctly I was just running the builder using a query with invalid syntax and another query with an object that doesn't exist in the org.